### ISSUE TITLE
High retention on Prometheus

### ISSUE THREAD

**Issue by metalmatze** [improbable-eng/thanos#283]
In the example the retention for Prometheus itself is set to 24h.
When setting this to, for example, 1 week (168h) does Thanos query the object storage less for that time range?

**Comment by Bplotka** [comment]
Hello!
Good question. (: 

**TL;DR:**  No.  For Thanos query it does not matter if the underlying Store API is store, sidecar or ruler.

This is related discussion: https://github.com/improbable-eng/thanos/issues/82 I remember I started doing some implementation of it, but essentially this is bit fuzzy problem. Basically:
- you don't know how many stores you have in the system. We aim support one per bucket for now, but you can have multiple buckets.
- you don't really know what "sources" (what metrics) store node has access too.
- races can happen (although that could be mitigated by some time margin).
- we are not really sure if it's "better" to ask store or Prometheus (sidecar/source). Better means cheaper? faster? simpler? more resilient?
